apps/els_lsp/src/els_dt_signatures.erl

```diff
@@ -72,11 +73,18 @@ insert(Map) when is_map(Map) ->
   els_db:write(name(), Record).
 
 -spec lookup(mfa()) -> {ok, [item()]}.
-lookup(MFA) ->
+lookup({M, _F, _A} = MFA) ->
+  {ok, _Uris} = els_utils:find_modules(M),
   {ok, Items} = els_db:lookup(name(), MFA),
   {ok, [to_item(Item) || Item <- Items]}.
 
--spec delete_by_module(atom()) -> ok.
-delete_by_module(Module) ->
-  Pattern = #els_dt_signatures{mfa = {Module, '_', '_'}, _ = '_'},
-  ok = els_db:match_delete(name(), Pattern).
+-spec delete_by_uri(uri()) -> ok.
+delete_by_uri(Uri) ->
+  case filename:extension(Uri) of
+    <<".erl">> ->
```

Comment:
Signatures are currently indexed by `MFA` and the respective info is indexed while the module itself is indexed (taking into account the chain of header files). The reason why the check is here is that we also process headers for indexing and that caused an issue since it was deleting all, say, `file` specs when indexing `file.hrl`, which is of course incorrect. Indexing of headers should be reconsidered, but I would keep that out of scope for this PR.